bundles/org.eclipse.mita.program.ui/src/org/eclipse/mita/program/ui/labeling/ProgramDslLabelProvider.xtend

```diff
@@ -228,4 +228,12 @@ class ProgramDslLabelProvider extends DefaultEObjectLabelProvider {
 		}
 		return super.convertToStyledString(text)
 	}
+
+	def protected dispatch getTypeName(TypedElement it) {
+		(BaseUtils.getType(it) ?: typeSpecifier)?.toString
+	}
+
+	def protected dispatch getTypeName(SystemResourceSetup it) {
+		(BaseUtils.getType(it) ?: type)?.toString
```

Comment:
I'm not sure you can always get `it.type`. It would probably be safer to use the Ecore model here:
```suggestion
		(BaseUtils.getType(it) ?: BaseUtils.getText(it, ProgramPackage.eINSTANCE.systemResourceSetup_Type))?.toString
```

(`BaseUtils.getText` tries to get the requested feature via `eGet`, and if that fails, get's the original source code)